crates/rpc/rpc-types/src/eth/fee.rs

```diff
@@ -20,3 +23,29 @@ pub struct FeeHistory {
     #[serde(default)]
     pub reward: Option<Vec<Vec<U256>>>,
 }
+
+/// LRU cache for `eth_feeHistory` RPC method. Block Number => Fee History.
+#[derive(Clone, Debug)]
+pub struct FeeHistoryCache(pub Arc<Mutex<LruCache<BlockNumber, FeeHistoryCacheItem>>>);
```

Comment:
perhaps we can improve this in a followup,
since fee history is used to estimate calls, I think we should try to cache the range `highest block - 2028..highest block`

so perhaps we can use a BtreeMap here instead and only insert missing blocks if there in the range?

Reply:
But I guess this is already kinda ensured because we only allow a certain block range? 
but not sure if this will deterministically evict only out-of-range blocks.

Reply:
> but not sure if this will deterministically evict only out-of-range blocks.

it should, because the most frequent use case for the `eth_feeHistory` call is `latest-BLOCK_COUNT..latest`, so we will move all these blocks to the head of the LRU cache evicting older ones like `latest-10-BLOCK_COUNT..latest-BLOCK_COUNT`